source/slang/slang-ir-peephole.cpp

```diff
@@ -250,6 +250,24 @@ struct PeepholeContext : InstPassBase
 
         switch (inst->getOp())
         {
+        case kIROp_AlignOf:
+            // Fold all calls to alignOf<T>() that returns a simple integer value.
+            if (targetProgram && inst->getDataType()->getOp() == kIROp_IntType)
+            {
+                IRSizeAndAlignment sizeAlignment;
+                auto alignOfInst = as<IRAlignOf>(inst);
+                auto baseType = alignOfInst->getBaseOp()->getDataType();
+                getNaturalSizeAndAlignment(targetProgram->getOptionSet(), baseType, &sizeAlignment);
```

Comment:
If this failed, should break.

Reply:
the subsequent check helps with that I guess, but I added a conditional check around getNaturalSizeAndAlignment